codyze-specification-languages/coko/coko-dsl/src/test/kotlin/de/fraunhofer/aisec/codyze/specificationLanguages/coko/dsl/CokoSarifBuilderTest.kt

```diff
@@ -0,0 +1,282 @@
+            includeBlocklist = listOf(),
+            includePaths = listOf(),
+            includeAllowlist = listOf(),
+            loadIncludes = false,
+            passes = listOf(UnreachableEOGPass::class, EdgeCachePass::class),
+        )
+
+    val cokoRulewithoutRuleAnnotation = object : CokoRule {
+        override val annotations: List<Annotation>
+            get() = emptyList()
+        override val isAbstract: Boolean
+            get() = TODO("Not yet implemented")
+        override val isExternal: Boolean
+            get() = TODO("Not yet implemented")
+        override val isFinal: Boolean
+            get() = TODO("Not yet implemented")
+        override val isInfix: Boolean
+            get() = TODO("Not yet implemented")
+        override val isInline: Boolean
+            get() = TODO("Not yet implemented")
+        override val isOpen: Boolean
+            get() = TODO("Not yet implemented")
+        override val isOperator: Boolean
+            get() = TODO("Not yet implemented")
+        override val isSuspend: Boolean
+            get() = TODO("Not yet implemented")
+        override val name: String
+            get() = "norule"
+        override val parameters: List<KParameter>
+            get() = TODO("Not yet implemented")
+        override val returnType: KType
+            get() = TODO("Not yet implemented")
+        override val typeParameters: List<KTypeParameter>
+            get() = TODO("Not yet implemented")
+        override val visibility: KVisibility?
+            get() = TODO("Not yet implemented")
+
+        override fun call(vararg args: Any?): Evaluator {
+            TODO("Not yet implemented")
+        }
+
+        override fun callBy(args: Map<KParameter, Any?>): Evaluator {
+            TODO("Not yet implemented")
+        }
+    }
+
+    @Test
+    fun `test empty rules list causes empty reportingDescriptors list`() {
+        val backend = CokoCpgBackend(cpgConfiguration)
+        val csb = CokoSarifBuilder(rules = emptyList(), backend = backend)
+
+        assertTrue(csb.reportingDescriptors.isEmpty())
+    }
+
+    @Test
+    fun `test spec without rule annotation`() {
+        val backend = CokoCpgBackend(cpgConfiguration)
+        val csb = CokoSarifBuilder(rules = listOf(cokoRulewithoutRuleAnnotation), backend = backend)
+
+        val reportingDescriptor = csb.reportingDescriptors.first()
+        assertNotNull(reportingDescriptor)
+        assertNull(reportingDescriptor.shortDescription)
+        assertNull(reportingDescriptor.fullDescription)
+        assertNull(reportingDescriptor.defaultConfiguration)
+        assertNull(reportingDescriptor.help)
+        assertNull(reportingDescriptor.properties)
+    }
+
+    @Test
+    fun `test rule with default shortDescription`() {
+        val specFiles = listOfNotNull(
+            CokoSarifBuilderTest::class.java.classLoader
+                .getResource("sarif/ruledefaults.codyze.kts")
+        ).map { Path(it.path) }
```

Comment:
Path(it.path) fails in with an InvalidPathException in Windows because the path starts with "/C:/..." where the semicolon after the leading slash is marked as an invalid character.
it.toURI().toPath() circumvents the issue.